include/mbgl/map/map.hpp

```diff
@@ -154,7 +154,9 @@ class Map : private util::noncopyable {
     void removeSource(const std::string& sourceID);
 
     // Layers
+    std::vector<style::Layer*> getLayers();
     style::Layer* getLayer(const std::string& layerID);
+    void setLayers(std::vector<style::Layer*>&);
```

Comment:
`Map`/`Style` must have ownership of the layers, so this signature needs to be `void setLayers(std::vector<std::unique_ptr<style::Layer>>)`.